fred2/addmodifycontainerdlg.cpp

```diff
@@ -800,24 +806,19 @@ void CAddModifyContainerDlg::OnBnClickedDeleteContainer()
 
 	char buffer[256];
 
-	// if the container has been renamed, search using the original name
+	// get both current and original names, even if identical
 	const SCP_string curr_name = get_current_container().container_name;
-	SCP_string orig_name;
-	SCP_string name_to_check;
-	const auto prev_name_it = m_new_to_old_names.find(curr_name);
-	if (prev_name_it != m_new_to_old_names.end()) {
-		orig_name = prev_name_it->second;
-		name_to_check = orig_name;
-	} else {
-		name_to_check = curr_name;
-	}
+	const auto orig_name_it = m_new_to_old_names.find(curr_name);
+	Assert(orig_name_it != m_new_to_old_names.end());
+	const SCP_string orig_name = orig_name_it->second;
 
-	const int times_used = m_p_sexp_tree->get_container_usage_count(name_to_check.c_str());
+	const int times_used =
+		is_null_container_name(orig_name) ? 0 : m_sexp_tree.get_container_usage_count(orig_name);
 	Assert(times_used >= 0);
 
 	if (times_used > 0) {
 		SCP_string container_name_str = "'" + curr_name + "'";
-		if (!orig_name.empty()) {
+		if (curr_name != orig_name) {
```

Comment:
Should this be a case-insensitive check?

Reply:
Hmm, good question. Since this is purely for a user-facing message, I'd say it depends on what makes for the best UX.

If the FREDder renamed a container from "foobar" to "FooBar" and then attempted to delete "FooBar" before closing the dialog, even though the container is currently in use, I can't decide whether it would be helpful to remind the FREDder that the name was originally "foobar". I don't know whether it would make the fact that container names are case-insensitive clearer or more confusing.

Thoughts?

Reply:
Oh, I see what you mean.  Hmm.

Ok, in this case I think a case-sensitive check is appropriate.  It's a change that the user made, even if it is treated as equivalent.